File: src/FoodOrderSystem.Persistence.MongoDB/DishModel.cs
using System;
using System.Collections.Generic;

namespace FoodOrderSystem.Persistence.MongoDB
{
    public class DishModel
    {
        public Guid Id { get; set; }
        public Guid RestaurantId { get; set; }
        public Guid CategoryId { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string ProductInfo { get; set; }
        public int OrderNo { get; set; }
        public IList<DishVariantModel> Variants { get; set; }        
    }
}

File: src/Gastromio.App/ClientApp/src/app/shared/models/dish.model.ts
import { DishVariantModel } from './dish-variant.model';

export class DishModel {
  constructor(init?: Partial<DishModel>) {
    if (init) {
      Object.assign(this, init);
    }
    this.variants = this.variants?.map(
      (variant) => new DishVariantModel(variant)
    );
  }

  public id: string;

  public name: string;

  public description: string;

  public productInfo: string;

  public variants: Array<DishVariantModel>;

  public clone(): DishModel {
    return new DishModel({
      id: this.id,
      name: this.name,
      description: this.description,
      productInfo: this.productInfo,
      variants: this.variants?.map((variant) => variant?.clone()),
    });
  }
}
